libs/vkd3d/device.c

```diff
@@ -7166,6 +7187,52 @@ void d3d12_device_mark_as_removed(struct d3d12_device *device, HRESULT reason,
     device->removed_reason = reason;
 }
 
+VkPipeline d3d12_device_get_or_create_vertex_input_pipeline(struct d3d12_device *device,
+        const struct vkd3d_vertex_input_pipeline_desc *desc)
+{
+    struct vkd3d_vertex_input_pipeline pipeline, *entry;
+
+    pthread_mutex_lock(&device->mutex);
```

Comment:
Should we use RW lock here similar to fallback pipeline cache? Using the full device mutex seems a bit pessimistic.

Reply:
Can do, correctly managing device-scope objects like that is just rather annoying.

Reply:
done.